paperqa/llms.py

```diff
@@ -87,17 +87,23 @@ def process_llm_config(
 
 
 async def embed_documents(
-    client: AsyncOpenAI, texts: list[str], embedding_model: str
+    client: AsyncOpenAI, texts: list[str], embedding_model: str, batch_size: int = 16
 ) -> list[list[float]]:
     """Embed a list of documents with batching."""
     if client is None:
         raise ValueError(
             "Your client is None - did you forget to set it after pickling?"
         )
-    response = await client.embeddings.create(
-        model=embedding_model, input=texts, encoding_format="float"
-    )
-    return [e.embedding for e in response.data]
+    N = len(texts)
+    embeddings = []
+    for i in range(0, N, batch_size):
+        response = await client.embeddings.create(
+            model=embedding_model,
+            input=texts[i : i + batch_size],
```

Comment:
Aside: Python 3.12 has [`itertools.batched`](https://docs.python.org/3/library/itertools.html#itertools.batched) to make this easier, but let's just stick with this for now

Reply:
Good to know - this will good to know in a few years once we get adoption.